packages/ui/src/pages/AdminPage.vue

```diff
@@ -1,28 +1,87 @@
 <script setup lang="ts">
 import { useRouter } from "vue-router";
 import { useAuthStore } from "@/stores/auth";
-import { useQuery } from "@tanstack/vue-query";
-import adminAPI from "@/api/admin";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/vue-query";
+import taskAPI from "@/api/tasks";
+import AppButton from "@/components/ui/AppButton.vue";
+import { ElNotification } from "element-plus";
 
 const router = useRouter();
 const authStore = useAuthStore();
+const queryClient = useQueryClient();
 
 if (!authStore.currentUser?.isAdmin) {
   router.push("/");
 }
 
+const deleteMutation = useMutation(taskAPI.deleteTask, {
+  onSuccess: () => {
+    queryClient.invalidateQueries(["tasks"]);
+    ElNotification({
+      title: "Success",
+      message: "Battery deleted successfully",
+      type: "success",
+    });
+  },
+});
+
+const uploadMutation = useMutation(taskAPI.uploadBattery, {
```

Comment:
NIT: Not super important, but you could use the `isLoading` field on both of these mutations to show a loading state while a battery is uploading/deleting (e.g. `v-loading="uploadMutation.isLoading || deleteMutation.isLoading"` somewhere in the template)